Pod/Classes/NJHNavigationBarSelectorPageViewController.m

```diff
@@ -0,0 +1,228 @@
+    CGFloat adjustedContentOffset = scrollView.contentOffset.x - CGRectGetWidth(self.view.frame);
+    
+    // This value represents the total x distance from the very left side of the first view controller being displayed by the page view controller to the left side of the view
+    // controller that is currently being displayed. This value stays constant as the user scrolls side to side and serves to differentiate (for example) between a left scroll
+    // from the first page to the second page and a left scroll from the second page to the third page, as in both of these cases, the adjustedContentOffset will go from 0 to the
+    // width of the page, making it impossible to tell them apart without some secondary number
+    NSInteger currentPageOffset = CGRectGetWidth(self.view.frame) * self.currentPageIndex;
+    
+    // For example, if the user was on the very last page of the view controller array, and we imagine the scroll view is set up as below, and the user starts swiping to the right,
+    // the currentPageOffset is the distance from the left side (marked LEFT) and the left side of the last view controller (marked CRT OFFSET). As the user swipes right, the adjustedContentOffset
+    // decreases from 0 to the negative width of the page. The finalOffset value calculated below effectively tracks the left side of the user's iphone screen's distance to the location marked
+    // LEFT below, allowing us to calculate how much to offset the selection view in the navigation bar
+    //
+    //
+    //                        |-------|  <--- This is the value of the adjusted offset (in this case it is a negative number)
+    //
+    //                        /-IPHONE SCREEN-\
+    // /-------------\ /------|------\ /------|------\
+    // |             | |      F      |C|      |      |
+    // |             | |      I      |R|      |      |
+    // |L            | |      N      |T|      |      |
+    // |E            | |      |      | |  PAGE CURR  |
+    // |F            | |      O      |O|  DISPLAYED  |
+    // |T            | |      F      |F|      |      |
+    // |             | |      S      |S|      |      |
+    // |             | |      E      |E|      |      |
+    // |             | |      T      |T|      |      |
+    // \-------------/ \------|------/ \------|------/
+    //                        \---------------/
+    CGFloat finalOffset = currentPageOffset + adjustedContentOffset;
+    
+    // Since the final offset is the distance from the "left" most view in the diagram above to the left side of the screen, it can be converted into a fraction that
+    // describes how far the user has scrolled from side to side by dividing by the total length of the scroll view. Once we have this fraction, we multiply by the length
+    // of the bacground selection view to figure out how far we should offset the selection view to properly represent the scrolling that has happened on screen
```

Comment:
Haha, TLDR.

But I know the problem you are describing, so thanks for documenting! 

![Alt Text](https://media4.giphy.com/media/OJE1vnBGMAkyk/200.gif)